src/mixins/filing-template-mixin.ts

```diff
@@ -65,7 +66,10 @@ export default class FilingTemplateMixin extends Vue {
             extension: this.stateModel?.defineCompanyStep?.businessContact?.extension || ''
           },
           parties: this.stateModel?.addPeopleAndRoleStep?.orgPeople || [],
-          shareClasses: this.stateModel?.createShareStructureStep?.shareClasses || []
+          shareClasses: this.stateModel?.createShareStructureStep?.shareClasses || [],
+          incorporationAgreement: {
+            agreementType: this.stateModel.incorporationAgreementStep.agreementType
```

Comment:
Do you need a fallback value here if the user saves a draft without setting the agreement type?

Reply:
The default value is initialized as null. The schema checks against an enum. For draft filing if it is not selected it will be null and api will not validate. For submission the API will throw an error if it is null or not in the enum that has values sample, custom. So this code needs no change